src/mgmt/srv_system.c

```diff
@@ -631,7 +671,7 @@ ds_mgmt_join_handler(struct mgmt_join_in *in, struct mgmt_join_out *out)
 			goto out_lock;
 		}
 		D_DEBUG(DB_TRACE, "rank %u rejoined\n", rank);
-		notify_map_distributord(svc);
+		ds_rsvc_request_map_dist(&svc->ms_rsvc);
```

Comment:
the out->jo_rank assignment on line 658 and this ds_rsvc_request_map_dist() are included in some of the steps that also appear just before the out_lock: label. Can another label be added just before out_lock: that makes this ds_rsvc_request_map_dist() call - and goto that label here? - so that ds_rsvc_request_map_dist() is called on one line of code in this function? I realize the out->jo_flags assignment there is different than what is done here and so the new label placement would have to be carefully done relative to jo_flags and other assignments.